File: app/src/main/java/com/kinwatt/powermeter/data/SensorData.java
package com.kinwatt.powermeter.data;

import java.util.ArrayList;
import java.util.List;

public class SensorData {
    private String name;
    private String address;
    private List<ServiceData> services;

    public SensorData() {
        services = new ArrayList<>();
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }
    public void setAddress(String address) {
        this.address = address;
    }

    public List<ServiceData> getServices() {
        return services;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof  SensorData) {
            SensorData other = (SensorData)obj;
            return this.address.equals(other.address);
        }
        else  {
            return false;
        }
    }
}

File: app/src/main/java/com/kinwatt/powermeter/data/SensorData.kt
package com.kinwatt.powermeter.data

import com.kinwatt.powermeter.data.mappers.UUIDSerializer
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class SensorData(var name: String? = null, var address: String? = null) {

    val services: MutableList<ServiceData> = ArrayList()

    override fun equals(other: Any?) = if (other is SensorData) equals(other) else false

    fun equals(other: SensorData) = this.address == other.address
}

@Serializable
data class ServiceData(@Serializable(with= UUIDSerializer::class)var uuid: UUID)